spec/bundler_inject_spec.rb

```diff
@@ -143,6 +143,30 @@
           end
         end
 
+        it "with a filename path in a gem_path location " do
+          with_path_based_gem("https://github.com/rubyworks/ansi", "the_gem") do |path|
+            write_bundler_d_file <<~F
+              override_gem "ansi", :path => "the_gem"
+            F
+            bundle(:update, :env => {"BUNDLE_BUNDLER_INJECT__GEM_PATH" => path.dirname.to_s})
```

Comment:
Can you add a spec that shows multiple paths in the ENV var, and then something appearing in _not_ the first path?  This would exercise the File::PATH_SEPARATOR code as well as that the map picks a later value.